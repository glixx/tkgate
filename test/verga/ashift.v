module top;
	reg [127: 0] big;
	reg [31:0] i;
	integer j;

	initial
		begin
			$display("Starting ashift test");

			i = 32'hdeadbeef;
			for (j=0; j<8; j = j + 1)
				begin
					$display("i = %08h", i);
					i = i >>> 4;
				end

			big = 128'h8123456789abcdefdeadbeefcafebabe;
			for (j=0; j<32; j = j + 1)
				begin
					$display("big = %032h", big);
					big = big >>> 4;
				end

			big = 128'h8123456789abcdefdeadbeefcafebabe;
			$display("big>>>4   = %032h", big>>>4);
			$display("big>>>32  = %032h", big>>>32);
			$display("big>>>39  = %032h", big>>>39);
			$display("big>>>64  = %032h", big>>>64);
			$display("big>>>68  = %032h", big>>>68);
			$display("big>>>200 = %032h", big>>>200);


		end

endmodule
