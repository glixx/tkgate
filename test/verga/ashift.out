Starting ashift test
i = deadbeef
i = fdeadbee
i = ffdeadbe
i = fffdeadb
i = ffffdead
i = fffffdea
i = ffffffde
i = fffffffd
big = 8123456789abcdefdeadbeefcafebabe
big = f8123456789abcdefdeadbeefcafebab
big = ff8123456789abcdefdeadbeefcafeba
big = fff8123456789abcdefdeadbeefcafeb
big = ffff8123456789abcdefdeadbeefcafe
big = fffff8123456789abcdefdeadbeefcaf
big = ffffff8123456789abcdefdeadbeefca
big = fffffff8123456789abcdefdeadbeefc
big = ffffffff8123456789abcdefdeadbeef
big = fffffffff8123456789abcdefdeadbee
big = ffffffffff8123456789abcdefdeadbe
big = fffffffffff8123456789abcdefdeadb
big = ffffffffffff8123456789abcdefdead
big = fffffffffffff8123456789abcdefdea
big = ffffffffffffff8123456789abcdefde
big = fffffffffffffff8123456789abcdefd
big = ffffffffffffffff8123456789abcdef
big = fffffffffffffffff8123456789abcde
big = ffffffffffffffffff8123456789abcd
big = fffffffffffffffffff8123456789abc
big = ffffffffffffffffffff8123456789ab
big = fffffffffffffffffffff8123456789a
big = ffffffffffffffffffffff8123456789
big = fffffffffffffffffffffff812345678
big = ffffffffffffffffffffffff81234567
big = fffffffffffffffffffffffff8123456
big = ffffffffffffffffffffffffff812345
big = fffffffffffffffffffffffffff81234
big = ffffffffffffffffffffffffffff8123
big = fffffffffffffffffffffffffffff812
big = ffffffffffffffffffffffffffffff81
big = fffffffffffffffffffffffffffffff8
big>>>4   = f8123456789abcdefdeadbeefcafebab
big>>>32  = ffffffff8123456789abcdefdeadbeef
big>>>39  = ffffffffff02468acf13579bdfbd5b7d
big>>>64  = ffffffffffffffff8123456789abcdef
big>>>68  = fffffffffffffffff8123456789abcde
big>>>200 = ffffffffffffffffffffffffffffffff
